tests/PhpUnit/SymfonyAssertsTraitTest.php

```diff
@@ -0,0 +1,152 @@
+    use SymfonyAssertsTrait;
+
+    /**
+     * @var SchemaManager
+     */
+    protected $schemaManager;
+
+    protected function setUp()
+    {
+        $this->schemaManager = new SchemaManager('file://' . __DIR__ . '/../fixture/petstore-with-external-docs.json');
+    }
+
+    public function testAssertResponseMatch()
+    {
+        $response = $this->createResponse(200, $this->getValidHeaders(), $this->getValidResponseBody());
+
+        self::assertResponseMatch($response, $this->schemaManager, '/api/pets', 'get');
+    }
+
+    public function testAssertResponseAndRequestMatch()
+    {
+        $response = $this->createResponse(200, $this->getValidHeaders(), $this->getValidResponseBody());
+        $request = $this->createReqyest('GET', '/api/pets');
+
+        self::assertResponseAndRequestMatch($response, $request, $this->schemaManager);
+    }
+
+    public function testAssertResponseBodyDoesNotMatch()
+    {
+        $response = <<<JSON
+[
+  {
+    "id": 123456789
+  }
+]
+JSON;
+        $response = $this->createResponse(200, $this->getValidHeaders(), $response);
+
+        try {
+            self::assertResponseMatch($response, $this->schemaManager, '/api/pets', 'get');
+            self::fail('Expected ExpectationFailedException to be thrown');
+        } catch (ExpectationFailedException $e) {
+            self::assertRegExp(
+                sprintf('/^Failed asserting that \[{.+}\] is valid.\\n\[%1$s\] The property %1$s is required\\n$/i', 'name'),
+                $e->getMessage()
+            );
+        }
+    }
+
+    public function testAssertResponseMediaTypeDoesNotMatch()
+    {
+        $response = $this->createResponse(
+            200,
+            ['Content-Type' => ['application/pdf; charset=utf-8']],
+            $this->getValidResponseBody()
+        );
+
+        try {
+            self::assertResponseMatch($response, $this->schemaManager, '/api/pets', 'get');
+            self::fail('Expected ExpectationFailedException to be thrown');
+        } catch (ExpectationFailedException $e) {
+            self::assertEquals(
+                "Failed asserting that 'application/pdf' is an allowed media type (application/json, application/xml, text/xml, text/html).",
+                $e->getMessage()
+            );
+        }
+    }
+
+    public function testAssertResponseHeaderDoesNotMatch()
+    {
+        $headers = [
+            'Content-Type' => ['application/json'],
+            // 'ETag' => ['123'], // Removed intentional
+        ];
+
+        $response = $this->createResponse(200, $headers, $this->getValidResponseBody());
+
+        try {
+            self::assertResponseMatch($response, $this->schemaManager, '/api/pets', 'get');
+            self::fail('Expected ExpectationFailedException to be thrown');
+        } catch (ExpectationFailedException $e) {
+            self::assertRegExp(
+                sprintf('/^Failed asserting that {.+} is valid.\\n\[%1$s\] The property %1$s is required\\n$/i', 'etag'),
```

Comment:
It's preferible assert the complete string because this library MUST return useful messages to the developer.